docs/en/engines/table-engines/integrations/rabbitmq.md

```diff
@@ -0,0 +1,137 @@
+---
+toc_priority: 6
+toc_title: RabbitMQ
+---
+
+This engine works with [RabbitMQ](https://www.rabbitmq.com).
+
+RabbitMQ lets you:
+-   Publish or subscribe to data flows.
+-   Process streams as they become available.
+
+## Creating a Table {#table_engine-rabbitmq-creating-a-table}
+
+``` sql
+CREATE TABLE [IF NOT EXISTS] [db.]table_name [ON CLUSTER cluster]
+(
+    name1 [type1] [DEFAULT|MATERIALIZED|ALIAS expr1],
+    name2 [type2] [DEFAULT|MATERIALIZED|ALIAS expr2],
+    ...
+) ENGINE = RabbitMQ SETTINGS
+    rabbitmq_host_port = 'host:port',
+    rabbitmq_exchange_name = 'exchange_name',
+    rabbitmq_format = 'data_format'[,]
+    [rabbitmq_exchange_type = 'exchange_type',]
+    [rabbitmq_routing_key_list = 'key1,key2,...',]
+    [rabbitmq_row_delimiter = 'delimiter_symbol',]
+    [rabbitmq_num_consumers = N,]
+    [rabbitmq_num_queues = N,]
+    [rabbitmq_transactional_channel = 0]
+```
+
+Required parameters:
+
+-   `rabbitmq_host_port` – host:port (for example, `localhost:5672`).
+-   `rabbitmq_exchange_name` – RabbitMQ exchange name.
+-   `rabbitmq_format` – Message format. Uses the same notation as the SQL `FORMAT` function, such as `JSONEachRow`. For more information, see the [Formats](../../../interfaces/formats.md) section.
+
+Optional parameters:
+
+-   `rabbitmq_exchange_type` – The type of RabbitMQ exchange: `direct`, `fanout`, `topic`, `headers`, `consistent-hash`. Default: `fanout`.
+-   `rabbitmq_routing_key_list` – A comma-separated list of routing keys.
+-   `rabbitmq_row_delimiter` – Delimiter character, which ends the message.
+-   `rabbitmq_num_consumers` – The number of consumers per table. Default: `1`. Specify more consumers if the throughput of one consumer is insufficient.
+-   `rabbitmq_num_queues` – The number of queues per consumer. Default: `1`. Specify more queues if the capacity of one queue per consumer is insufficient. Single queue can contain up to 50K messages at the same time.
+-   `rabbitmq_transactional_channel` –  Wrap insert queries in transactions. Default: `0`.
+
+Examples:
+
+``` sql
+  CREATE TABLE queue (
+    key UInt64,
+    value UInt64
+  ) ENGINE = RabbitMQ SETTINGS rabbitmq_host_port = 'localhost:5672',
+                            rabbitmq_exchange_name = 'exchange1',
+                            rabbitmq_format = 'JSONEachRow',
+                            rabbitmq_num_consumers = 5;
+
+  SELECT * FROM queue LIMIT 5;
+
+  CREATE TABLE queue2 (
+    key UInt64,
+    value UInt64
+  ) ENGINE = RabbitMQ('localhost:5672', 'key1', 'exchange1', 'JSONEachRow', '\\n');
+
+```
+
+<details markdown="1">
+
+<summary>Deprecated Method for Creating a Table</summary>
```

Comment:
let's just drop it, no need to advertise how to use new features with legacy syntax

Reply:
+1